### ISSUE TITLE
Implement RAdam optimizer ? 

### ISSUE THREAD

**Issue by HareshMiriyala** [pytorch/pytorch#24892]
## 🚀 Feature
implement RAdam optimizer ? 

## Motivation
I've recently come across this paper on rectified Adam, which shows a clear improvement on the existing Adam optimizer. There is a issue on the tensorflow github page for a feature request and I thought pytorch definitely needs someone to bring it up. 

## Pitch
Being able to use RAdam in Pytorch. 

## Additional context

RAdam Paper : https://arxiv.org/pdf/1908.03265v1.pdf

cc @vincentqb

**Comment by HareshMiriyala** [comment]
github repo from Liyuan : https://github.com/LiyuanLucasLiu/RAdam

**Comment by Tanmay-Jaiswal** [comment]
I haven't contributed to pytorch before but I have been using it for almost year.  I'd like to take up this issue if possible. Do you think this issue is doable for a first-timer?

**Comment by mostafaelhoushi** [comment]
> I haven't contributed to pytorch before but I have been using it for almost a year. I'd like to take up this issue if possible. Do you think this issue is doable for a first-timer?

I am not 100% sure but I think it is doable. I see the repo from Liyuan simply creates RAdam by extending PyTorch's Optimizer's class:
https://github.com/LiyuanLucasLiu/RAdam/blob/master/radam.py

I think placing this file in the correct place in the PyTorch repo and running test cases will be enough.

**Comment by Tony-Y** [comment]
https://github.com/pytorch/pytorch/pull/28401#issuecomment-549090571

**Comment by gchanan** [comment]
@vincentqb: can you make a call on this and remove needs research?

**Comment by vincentqb** [comment]
We don't think this method should be in pytorch core, at least not yet.

Our reservation is that we want to include methods that the community uses as a standard, or else the code maintenance problem balloons up for us. We do show discretion based on what the paper shows as evidence, for example BatchNorm was included within weeks of it's publish date (in Torch). In terms of rejected methods, we've rejected newly minted papers such as Swish (#3260, #3182), [Yellowfin](https://github.com/pytorch/pytorch/issues/1960) and many others that haven't become standardized in the community (like LSTM/Transformer/BatchNorm).

If you have a different opinion, let us know why, and we can re-evaluate.

**Comment by RicCu** [comment]
@vincentqb  wasn't [torch contrib](https://github.com/pytorch/contrib) set up for these non-mainstream algorithm implementations? Wouldn't it be worth it to revive that project? Or alternatively increase support/discovery of community implementations via torch hub?

**Comment by vincentqb** [comment]
This is a bigger discussion about how to discover related projects, optimizers being one example. I'll cc @ailzhang for torchhub.

**Comment by ailzhang** [comment]
Hi w.r.t the discovery problem, torchhub can help on that. If you would like to add a `hubconf.py` in https://github.com/LiyuanLucasLiu/RAdam it'll be already easily accessible for users. 
Also if you want to raise awareness of this new optimizer, feel free to send a PR in pytorch/hub so that it shows up https://pytorch.org/hub/. UI is mainly designed for models, but we could extend it a little I think.

**Comment by brando90** [comment]
so is there a pytorch approved implementation of RAdam or no?

**Comment by brando90** [comment]
> github repo from Liyuan : https://github.com/LiyuanLucasLiu/RAdam

@vincentqb what is wrong with that implementation?
